Float.TinCan/RemoteLRS.cs

```diff
@@ -494,10 +494,12 @@ public async Task<AgentProfileLRSResponse> RetrieveAgentProfile(string id, Agent
         }
 
         /// <inheritdoc/>
-        public Task<LRSResponse> SaveAgentProfile(AgentProfileDocument profile)
+        public async Task<LRSResponse> SaveAgentProfile(AgentProfileDocument profile)
         {
             Contract.Requires(profile != null);
-            return SaveAgentProfile(profile, RequestType.put);
+            var currentProfile = await RetrieveAgentProfile(profile.id, profile.agent);
+            profile.etag = currentProfile?.content?.etag;
+            return await SaveAgentProfile(profile, currentProfile == null ? RequestType.put : RequestType.post);
```

Comment:
@bbeversdorf IMHO, I think the implementing application is responsible for specifying the `etag`...and it looks like there is already a separate public method (albeit with a questionable name) for using `POST` vs `PUT` (`ForceSaveAgentProfile`).

Reply:
When the SPT was "resaving" the profile and there was already an entry, then we need to POST. This implementation of TinCan _always_ assumes it should be PUT which fails.

Reply:
I'm not sure what the stack trace is leading up to this (maybe this is a change that needs to be made in LocalLRSServer?), but the `ForceSaveAgentProfile` method appears to be the intended way to use `POST`.

Reply:
Initial agent profiles should be `PUT`. We would still need to know if there was a profile or not.

Reply:
> We would still need to know if there was a profile or not.

I still think that has to be the responsibility of the caller. The caller might be trying to persist outdated data here; so the caller needs to confirm that it knows the etag of the document it intends to be updating.

`PUT` can still be used for updating an existing document. `POST` is only used if you want to merge JSON documents.